storage-api/src/main/java/com/connexta/transformation/storage/api/status/TransformInfo.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api.status;
+
+import java.util.stream.Stream;
+
+/**
+ * A TransformInfo handles information about the entire process associated with a single transform
+ * request, i.e. info about an entire transformation.
+ */
+public interface TransformInfo extends TransformationStatus {
+
+  /**
+   * Adds the creation of a new metadata type as part of this transformation and persists an
+   * indication that the generation of that metadata is pending.
+   *
+   * @param metadataType the type of metadata to be generated (e.g. metacard, irm...)
+   * @return a {@link MetadataInfo} for the metadata to be generated
+   */
+  MetadataInfo add(String metadataType);
```

Comment:
✏️ Needs to throw `StorageException` since it's modifying the underlying datastore.